### PR TITLE
Spectral line

### PR DESCRIPTION
**Pull request by profxj**
REMOVED: Modified spectralline for AODM analysis.
REMOVED: Added a new test to test_absline_anly
Added ismatch function 
Added "closest" for SpectralLine to grab the closesline in the LineList

### CODE REVIEW

File: linetools/spectralline.py [hunk 1]
```diff
@@ -120,8 +124,18 @@ def cut_spec(self, normalize=False):
...
+            # Calculate
+            self.analy['spec'].velo = self.analy['spec'].relative_vel(
+                self.wrest*(1+self.attrib['z']))
+            # Cut
+            velo = self.analy['spec'].velo[pix] 
+            # Return
+            return fx, sig, velo
+        else:
+            # Return
+            return fx, sig, wave
```
Comment (nhmc) [inline linetools/spectralline.py]: I am not a fan of changing the return values based on a keyword flag.  Can you return a dictionary instead? i.e.:

```python
return fx, sig, dict(vel=velo, wave=wave)
```

or 

```python
return dict(fx=fx, sig=sig, vel=velo, wave=wave)
```

(I like the first option better)

File: linetools/spectralline.py [hunk 2]
```diff
@@ -242,46 +256,31 @@ def fill_data(self,trans, linelist=None):
                        } )
 
     # Perform AODM on the line
-    def aodm(self, **kwargs): 
+    def aodm(self, flg_sat=None):
```
Comment (nhmc) [inline linetools/spectralline.py]: Would it be better to pull this AODM code out of the spectral line class into its own function or module? Might be useful to be able to access it without have to create a SpectralLine.

File: linetools/spectralline.py [hunk 3]
```diff
@@ -291,8 +290,7 @@ def aodm(self, **kwargs):
             if len(lim) > 0:
                 sub = np.maximum(0.05, sig[satp[lim]]/5.)
                 nndt[satp[lim]] = np.log(1./sub)*cst
-                flg_sat = len(lim) 
-                raise ValueError('USE FLG_SAT')
+                flg_sat = True
```
Comment (nhmc) [inline linetools/spectralline.py]: `flg_sat` doesn't seem to be stored anywhere.

File: linetools/spectralline.py (hunk 1 above)
Comment (profxj) [inline linetools/spectralline.py]: Done

File: linetools/spectralline.py (hunk 3 above)
Comment (profxj) [inline linetools/spectralline.py]: AODM no longer incoporated in this PR

Comment (nhmc) [inline linetools/spectralline.py]: Ok great, please merge when you're ready.

**Pull request merged**